firebase_admin/firestore_async.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Cloud Firestore Async module.
+
+This module contains utilities for accessing the Google Cloud Firestore databases associated with
+Firebase apps. This requires the ``google-cloud-firestore`` Python module.
+"""
+
+try:
+    from google.cloud import firestore # pylint: disable=import-error,no-name-in-module
+    existing = globals().keys()
+    for key, value in firestore.__dict__.items():
+        if not key.startswith('_') and key not in existing:
+            globals()[key] = value
+except ImportError:
+    raise ImportError('Failed to import the Cloud Firestore library for Python. Make sure '
+                      'to install the "google-cloud-firestore" module.')
+
+from firebase_admin import _utils
+
+
+_FIRESTORE_ASYNC_ATTRIBUTE = '_firestore_async'
+
+
+def client(app=None):
+    """Returns a client that can be used to interact with Google Cloud Firestore.
+
+    Args:
+      app: An App instance (optional).
+
+    Returns:
+      google.cloud.firestore.Firestore_Async: A `Firestore Async Client`_.
```

Comment:
Yes. From what I've seen it is how links are referenced so that definitions are not too verbose. 
A similar example in the gapi firestore library [here](https://github.com/googleapis/python-firestore/blob/614254dd9024ca6751fa424c67c63e72397c573b/google/cloud/firestore_v1/types/write.py#L40).